### PR TITLE
stencil grapes plugin added to program-tools

### PR DESCRIPTION
**Pull request by 00salmon**
## Description of the change

> Tool for building a GrapesJS plugin template based on a stencil build. That plugin can then be published/plugged into grapesJS as a dependency.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: https://saasquatch.atlassian.net/browse/DEV-2995
 - Process.st launch checklist: https://app.process.st/checklists/Stencil-Grapes-Plugin-10-lPYrrGSerBykNYPmtaVPTQ

### Development

- [ ] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] This pull request has a Jira number
- [ ] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

File: packages/stencil-grapes-plugin/example/package.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+  "name": "grapes-test",
+  "version": "0.0.1",
```
Comment (00salmon) [inline packages/stencil-grapes-plugin/example/package.json:3]:
```suggestion
  "version": "1.0.0",
```

File: packages/stencil-grapes-plugin/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "@saasquatch/stencil-grapes-plugin",
+  "version": "0.1.2",
+  "description": "A stencil docs generator for generating GrapesJS plugins",
+  "source": "src/generator.ts",
+  "main": "dist/generator.js",
+  "module": "dist/generator.module.js",
+  "types": "dist/generator.d.ts",
```
Comment (loganvolkers) [inline packages/stencil-grapes-plugin/package.json:8]: @sam-beveridge please add a `repository` field

File: packages/stencil-grapes-plugin/src/generator.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+function tagValue(tags: JsonDocsTag[], name: string): string | undefined {
+  return tags.find((t) => t.name === name)?.text;
+}
+function jsonTagValue(tags: HasDocsTags, name: string) {
+  const value = tagValue(tags.docsTags, name);
+  return value && JSON.parse(value);
+}
+function hasTag(tagName: string) {
+  return (d: HasDocsTags) =>
+    d.docsTags?.find((t) => t.name === "tagName") ? true : false;
```
Comment (loganvolkers) [inline packages/stencil-grapes-plugin/src/generator.ts:126]: Is this the real thing?

```suggestion
    d.docsTags?.find((t) => t.name === tagName) ? true : false;
```

File: packages/stencil-grapes-plugin/package.json [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    "typescript": "^4.2.3"
+  },
+  "devDependencies": {
+    "@stencil/core": "^2.4.0",
+    "jest": "^26.6.3",
+    "microbundle": "^0.13.0"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/saasquatch/program-tools.git"
```
Comment (loganvolkers) [inline packages/stencil-grapes-plugin/package.json:28]: @sam-beveridge Needs `loction` field for `package/...`

**Review by loganvolkers — APPROVED** [review]
Looks good to me!

**Pull request merged**